### PR TITLE
FW-1340 FW-1307 Maintenance package (with workers) 

### PR DESCRIPTION
**Pull request by dy2288**
This code has the framework for executing jobs on a type - it is currently implementing category operations to migrate categories (moving code from external utils to internal code base).

### CODE REVIEW

File: FirstVoicesData/src/test/java/ca/firstvoices/runner/FirstVoicesDataFeature.java [hunk 1]
```diff
@@ -42,12 +42,13 @@
 @Deploy("org.nuxeo.ecm.platform.search.core")
 @Deploy("org.nuxeo.ecm.platform.webapp.types")
 
+// NOTE: All of FirstVoicesData may already be deployed? Not sure if this is needed.
```
Comment (davidlacho) [inline FirstVoicesData/src/test/java/ca/firstvoices/runner/FirstVoicesDataFeature.java:45]: I will make a ticket to look over unit tests deploys. These are getting out of hand.

Comment (davidlacho) [reply FirstVoicesData/src/test/java/ca/firstvoices/runner/FirstVoicesDataFeature.java:45]: https://firstvoices.atlassian.net/browse/FW-1369

File: FirstVoicesMaintenance/src/main/java/ca/firstvoices/dialect/RequiredJobsListener.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+              if (dialects == null || dialects.size() == 0) {
+                return;
+              }
+
+              AutomationService automation = Framework.getService(AutomationService.class);
+              MaintenanceLogger maintenanceLogger = Framework.getService(MaintenanceLogger.class);
+
+              for (DocumentModel dialect : dialects) {
+
+                Set<String> requiredJobs = maintenanceLogger.getRequiredJobs(dialect);
```
Comment (davidlacho) [inline FirstVoicesMaintenance/src/main/java/ca/firstvoices/dialect/RequiredJobsListener.java:57]: I like that in the future we might be able to do code re-use by making this loop in an abstract class, as discussed.

File: FirstVoicesData/src/test/java/ca/firstvoices/runner/FirstVoicesDataFeature.java (hunk 1 above)
Comment (dy2288) [reply FirstVoicesData/src/test/java/ca/firstvoices/runner/FirstVoicesDataFeature.java:45]: Thanks, also worth noting this package will need unit tests (and it probably won't be fun with all the async joy :D)

**Pull request merged**